docs/whirlpool/docs/02-Architecture Overview/07-TokenExtensions Support.md

```diff
@@ -0,0 +1,161 @@
+- MemoTransfer
+- MetadataPointer
+- TokenMetadata
+- ConfidentialTransfer (non-confidential transfer only)
+
+**Supported if it has initialized TokenBadge**
+- PermanentDelegate
+- TransferHook
+- MintCloseAuthority
+- DefaultAccountState (default state must be Initialized)
+
+> ℹ️ FreezeAuthority is not extensions, but it requires TokenBadge. 
+It should be disabled unless there is a reasonable reason (e.g., legal compliance). FreezeAuthority rejection does not apply to TokenProgram tokens, but does apply to TokenExtensions tokens. 
+
+**Not Supported**
+- InterestBearing -> supported at slot 294954598 (Oct 11, 2024 06:26:42 UTC)
+- Group, GroupPointer
+- Member, MemberPointer
+- NonTransferable
+- All other extensions not listed above
+
+## New Instructions
+### V2 Instructions
+V2 instructions have been added to handle tokens owned by Token-2022 program.
+
+- V1 instructions don’t work for pools with Token-2022 tokens.
+- V2 instructions work for pools with the following combinations. So V2 encompasses V1; an implementation that always uses V2 is simple, but the downside is increased transaction size if ALT is not used because of the large number of accounts required.
+    - Token / Token
+    - Token-2022 / Token
+    - Token / Token-2022
+    - Token-2022 / Token-2022
+- Token-2022 program requires Mint account for transfer, so the V2 instructions receive token programs and Mint accounts. It must also receive an SPL Memo program to support MemoTransfer.
+- When dealing with tokens that have TransferHook extension, instructions will receive the accounts used by the hook program as remaining accounts.
+- To reduce transfer fee, two_hop_swap_v2 transfers token directly from the first pool to the second pool. The user's token account is not passed through.
+#### For Trade
+|Instruction|Corresponding V1|Notes|
+|---|---|---|
+|swap_v2|swap||
+|two_hop_swap_v2|two_hop_swap|intermediate token accounts have been eliminated to reduce transfer fee overhead|
+
+#### For Liquidity
+|Instruction|Corresponding V1|
+|---|---|
+|increase_liquidity_v2|increase_liquidity|
+|decrease_liquidity_v2|decrease_liquidity|
+
+#### For Fees and Rewards
+|Instruction|Corresponding V1|
+|---|---|
+|collect_fees_v2|collect_fees|
+|collect_reward_v2|collect_reward|
+|collect_protocol_fees_v2|collect_protocol_fees|
+|set_reward_emissions_v2|set_reward_emissions|
+
+#### For Pool
+|Instruction|Corresponding V1|
+|---|---|
+|initialize_pool_v2|initialize_pool|
+|initialize_reward_v2|initialize_reward|
+
+### ConfigExtension & TokenBadge Instructions
+- initialize_config_extension
+- set_config_extension_authority
+- set_token_badge_authority
+- initialize_token_badge
+- delete_token_badge
+
+## Token Badge
+It was determined that the ability to freely create pools with tokens that have functions that may malfunction the pools, such as PermanentDelegate, has more disadvantages for abuse than advantages. However, several stable coins already have PermanentDelegate extension.
```

Comment:
> However, several stable coins already have PermanentDelegate extension.

Is this relevant here? I was expecting some comment like `To still support extensions that can potentially break whirlpools, the concept of TokenBadge was introduced`